doc/api/diagnostics_channel.md

```diff
@@ -0,0 +1,189 @@
+# Diagnostics Channel
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/diagnostics_channel.js -->
+
+The `diagnostics_channel` module provides an API to report diagnostic data
+via named channels. It can be accessed using:
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+```
+
+## Public API
+
+### Overview
+
+Following is a simple overview of the public API.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+// Subscribe to a channel by name
+diagnostics_channel.subscribe('my-channel', (name, data) => {
+  // Received data
+});
+
+// Subscribe to any channels matching a function result
+diagnostics_channel.subscribe({
+  shouldListen(name) {
+    return /^\s+-channel/.test(name);
+  },
+  onMessage(name, data) {
+    // Received data
+  }
+});
+
+// Subscribe to every channel
+diagnostics_channel.subscribe((name, data) => {
+  // Received data
+});
+
+// Get a reusable channel object
+const channel = diagnostics_channel.publisher('my-channel');
+
+// Subscribe directly to a channel object
+channel.subscribe((name, data) => {
+  // Received data
+});
+
+// Check if the channel has an active subscriber
+if (channel.shouldPublish()) {
```

Comment:
This could be just a getter.

Reply:
Yep, I just chose `shouldPublish()` to mirror the design of the original diagnostics_channel module from Microsoft. I also had `channel.active` in my initial prototype, which aligns better with the `channel.ifActive?` pattern.